src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -101,6 +101,12 @@ public class Connector {
     private static final OkHttpClient baseClient =
             JenkinsOkHttpClient.newClientBuilder(new OkHttpClient()).build();
 
+    private static final Map<GitHub, StandardCredentials> gitHubCredentialsMap = new ConcurrentHashMap<>();
```

Comment:
I would not store a long-lived reference to the `StandardCredentials` objects themselves. Instead I would create some kind of `record` that records the relevant metadata and only hold a reference to that record. You also need to make sure this map gets cleaned up when the clients are removed. The easiest thing to do in this case might be to use a `WeakHashMap`, so that removal of the clients from the other maps automatically removes them from this map too. Alternatively, you could update the code that handles cleanup to clean up this map directly.